### PR TITLE
feat: add UPE tracking capability

### PR DESCRIPTION
**Pull request by frosso**
## Changes proposed in this Pull Request:

Tracking events for enabling/disabling the UPE feature.

Background info: p1631646724053500-slack-C02BHNRN6R1

## Testing instructions
- Ensure you have "Enables WooCommerce Analytics" enabled at http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features (if available)
- Ensure you have "Allow usage of WooCommerce to be tracked" enabled at http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com
- Ensure you have the `_wcstripe_feature_upe_settings` feature enabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc_stripe-onboarding_wizard to enable UPE
- Enable UPE
- You should see a network request for the tracked event on the `pixel.wp.com` domain
- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe to disable UPE
- Disable UPE through the hamburger menu
- You should see a network request for the tracked event on the `pixel.wp.com` domain
Screen Shot 2021-09-14 at 6 08 11 PM

### CODE REVIEW

File: client/tracking/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import domReady from '@wordpress/dom-ready';
```
Comment (frosso) [inline client/tracking/index.js:4]: Same as what we're doing in WCPay: https://github.com/Automattic/woocommerce-payments/blob/538b322f4576b7cb070332dc627e8a21cfab5c7a/client/tracks/index.js

File: client/tracking/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * Records site event.
+ *
+ * @param {string}  eventName       Name of the event.
+ * @param {Object?} eventProperties Event properties.
+ */
+export function recordEvent( eventName, eventProperties ) {
+	// Wc-admin track script could be enqueued after our plugin, wrap in domReady
+	// to make sure we're not too early.
+	domReady( () => {
+		console.log( '###', { isEnabled: isEnabled(), lib: getLibrary() } );
```
Comment (waclawjacek) [inline client/tracking/index.js]: This line should probably go, right?

```suggestion
```

**Review by waclawjacek — CHANGES_REQUESTED** [review]
Works exactly as described! :100: Clicking request changes because of that `console.log` comment.

BTW for testing: turn off your adblocker and the request is an image rather than a XHR request. :)

File: client/tracking/index.js (hunk 2 above)
Comment (frosso) [reply client/tracking/index.js]: 🤦 how did I miss that! 😆

**Review by waclawjacek — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**